### PR TITLE
Make maximum exponential back-off retries configurable

### PR DESCRIPTION
**Pull request by n-hebert**
Hello! :wave: 

The new version works great! We do have a few environments that still fail due to heavy load, so I came up with a small change to fix that.

This will allow users to back off to arbitrary lengths, if they have heavy load environments.

### CODE REVIEW

**Comment by n-hebert** [comment]
I'm also thinking that the jitter of 500ms is too small since the "bucket" refills at 20 per second and 500ms leaves all of the envs in a matrix fairly close together. 

A 2000ms or even higher would help ungroup the calls.

**Comment by n-hebert** [comment]
Some AWS blog reading on this shows that we may want to look into "Decorrelated" or "Full" jitter alternatives to the current impl.
https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

**Comment by einaregilsson** [comment]
Sorry, haven't had time to look at this, will do so when I get back to Iceland in a few weeks.

**Comment by n-hebert** [comment]
No worries! We've taken it the changes submitted here for a test drive and they work. 

Implementing the jitter properly to the spec linked above should simply speed it up :smile:

File: aws-api-request.js [hunk 1]
```diff
@@ -138,7 +137,7 @@ function awsApiRequest(options, retryAttempt = 0) {
                     //2~8 * 100 = 25600
                     //2~9 * 100 = 51200
 
-                    if (retryAttempt > MAX_RETRY_COUNT) {
```
Comment (gzur) [inline aws-api-request.js:141]: This sort of makes the `MAX_RETRY_COUNT` variable useless. 

I think you can just delete it.

File: aws-api-request.js [hunk 2]
```diff
@@ -138,7 +137,7 @@ function awsApiRequest(options, retryAttempt = 0) {
...
+                    if (retryAttempt > maxBackoffRetries) {
                         //Give them the error result, the caller can then deal with it...
                         console.warn(`Retry attempt exceeded max retry count (${MAX_RETRY_COUNT})... Giving up...`);
```
Comment (gzur) [inline aws-api-request.js:142]: and use your new variable in the `console.warn`
```suggestion
                        console.warn(`Retry attempt exceeded max retry count (${maxBackoffRetries})... Giving up...`);
```

**Review by gzur — CHANGES_REQUESTED** [review]
Hi @n-hebert - I'm sorry to report that Einar passed away earlier this year. 

My name is Gissur and I have assumed stewardship over Einar's GitHub account on behalf of his widow.

---
### On to the PR
This looks good to me, mostly. 

Just some minor cleanup, which you can see in the comments.

File: aws-api-request.js (hunk 2 above)
Comment (n-hebert) [reply aws-api-request.js:142]: Done in #96 where I have write access :slightly_smiling_face:

**Comment by n-hebert** [comment]
Replaced by #96 since I don't have write-access to Rewind's fork

**Pull request closed without merging**